docs/developer/decision-records/2025-02-06-hashicorp-vault-authentication-refactor/README.md

```diff
@@ -0,0 +1,77 @@
+# HashiCorp Vault authentication refactor
+
+## Decision
+
+We will refactor the HashiCorp Vault implementation to support the usage of different methods of authentication.
+
+## Rationale
+
+The HashiCorp Vault supports multiple different authentication methods, but the current implementation supports
+only one and provides no way to customize or exchange authentication methods. The full list of supported
+methods can be found in the [HashiCorp Vault Documentation](https://developer.hashicorp.com/vault/docs/auth). Some examples are:
+
+* [Token auth method](https://developer.hashicorp.com/vault/docs/auth/token) (already supported)
+* [Kubernetes auth method](https://developer.hashicorp.com/vault/docs/auth/kubernetes)
+* [AppRole auth method](https://developer.hashicorp.com/vault/docs/auth/approle)
+
+## Approach
```

Comment:
I'm sort of missing a clear statement as to what the extension point will be and where and how downstream extensions can hook in. If i understand correctly, the `HashicorpVaultTokenProvider` will be that extension point.  If so, then the code sample below should be annotated with `@ExtensionPoint`

Reply:
Good point. Added `@ExtensionPoint` to the code sample and also included a sentence clearly stating this.